user/details/script.js

```diff
@@ -661,6 +665,260 @@ function showContent() {
+  } else if (diff >= 60 * 1000) {
+    return Math.floor(diff / (60 * 1000)) + ' minutes ago';
+  } else {
+    return Math.floor(diff / 1000) + ' seconds ago';
+  }
+}
+
+function formatDate(dateStr) {
+  const date = new Date(dateStr);
+  const options = {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+    hour: '2-digit',
+    minute: '2-digit',
+    timeZone: 'Asia/Kolkata',
+  };
+  const formattedDate = date.toLocaleDateString('en-US', options);
+  return formattedDate.split(',')[0];
+}
+
+function generatePRsTabDetails() {
+  const div = createElement({
+    type: 'div',
+    classList: ['hidden-content', 'hide'],
+  });
+  const prsSec = createElement({
+    type: 'div',
+    classList: ['user-pr'],
+  });
+  const pagination = createElement({ type: 'div', classList: ['pagination'] });
+  const prevBtn = createElement({
+    type: 'button',
+    classList: ['pagination-load-more'],
+  });
+  prevBtn.appendChild(createTextNode('Load More'));
+  prevBtn.addEventListener('click', loadMorePRs);
+  pagination.append(prevBtn);
+  div.append(prsSec, pagination);
+  document.querySelector('.accordion-prs').appendChild(div);
+}
+
+async function getUserPRs() {
+  try {
+    const res = await makeApiCall(
+      `${API_BASE_URL}/pullrequests/user/${username}`,
+    );
+    if (res.status === 200) {
+      const data = await res.json();
+      userAllPRs = data.pullRequests;
+      totalPRsPages = Math.ceil(userAllPRs.length / PRsPerPage);
+      const prs = getPRstoFetch(userAllPRs, currentPRPageIndex);
+      generatePRsTabDetails();
+      generateUserPRsList(prs);
+    }
+  } catch (err) {
+    console.log(err);
+  }
+}
+
+function getPRstoFetch(userPR, currentPRPageIndex) {
+  const startIndex = currentPRPageIndex * PRsPerPage - PRsPerPage;
+  const endIndex = currentPRPageIndex + PRsPerPage;
+  return userPR.filter((_, index) => index > startIndex && index < endIndex);
+}
+
+function generateUserPRsList(userPRs) {
+  if (!userPRs.length) {
+    const errorE1 = createElement({
+      type: 'p',
+      classList: ['error'],
+    });
+    const container = document.querySelector(
+      '.accordion-prs > .hidden-content',
+    );
+    container.innerHTML = '';
+    errorE1.appendChild(createTextNode('No PRs to show'));
+    container.appendChild(errorE1);
+  } else {
+    userPRs.forEach((pr) => {
+      const PRsCard = createSinglePRCard(pr);
+      document.querySelector('.user-pr').appendChild(PRsCard);
+    });
+    if (currentPRPageIndex === 1) {
+      document.querySelector('.pagination-load-more').disabled = true;
+    } else {
+      document.querySelector('.pagination-load-more').disabled = false;
+    }
+    if (currentPRPageIndex === totalPRsPages) {
+      document.querySelector('.pagination-load-more').disabled = true;
+    } else {
+      document.querySelector('.pagination-load-more').disabled = false;
+    }
+  }
+}
+
+function createSinglePRCard(PR) {
```

Comment:
Write logic for creating table row into a separate function.

Reply:
done